services/vaults/lib/vaults/roi.js

```diff
@@ -39,11 +39,10 @@ const calculateYearlyRoi = (
   blocksPerDay,
   decimals,
 ) => {
-  const pricePerFullShareDelta =
-    (currentValue - previousValue) / 10 ** decimals;
   const blockDelta = currentBlock - previousBlock;
-  const dailyRoi = (pricePerFullShareDelta / blockDelta) * 100 * blocksPerDay;
-  const yearlyRoi = dailyRoi * 365;
+  const returnSincePrevBlock = (currentValue - previousValue) / previousValue;
+  const days = blockDelta / blocksPerDay;
```

Comment:
The constants used in this calculation are incorrect in a post Muir Glacier network. Block times are and have been consistently ~13 seconds, yielding ~4.6 as the constant to be used in calculating `blocksPerHour` which is used downstream in calculating `blocksPerDay`, hence affecting the results. 

So, Line 21 should change from
```javascript
const blocksPerMinute = 4;
```
to 
```javascript
const blocksPerMinute = 4.6;
```

## Reference Muir Glacier Block Propagation
Date | Block Number | Block Time Propagation
-- | -- | --
1/1/2021 | 1609459200 | 13.06
1/2/2021 | 1609545600 | 12.94
1/3/2021 | 1609632000 | 13.15
1/4/2021 | 1609718400 | 13.08
1/5/2021 | 1609804800 | 12.97
1/6/2021 | 1609891200 | 13.17
1/7/2021 | 1609977600 | 13.22
1/8/2021 | 1610064000 | 12.96
1/9/2021 | 1610150400 | 13.20
1/10/2021 | 1610236800 | 13.01
1/11/2021 | 1610323200 | 13.03
1/12/2021 | 1610409600 | 12.99
1/13/2021 | 1610496000 | 12.99
1/14/2021 | 1610582400 | 13.33
1/15/2021 | 1610668800 | 12.89
1/16/2021 | 1610755200 | 13.09
1/17/2021 | 1610841600 | 13.09
1/18/2021 | 1610928000 | 13.03
1/19/2021 | 1611014400 | 13.06
1/20/2021 | 1611100800 | 12.99
1/21/2021 | 1611187200 | 13.16
1/22/2021 | 1611273600 | 13.12
1/23/2021 | 1611360000 | 13.00
1/24/2021 | 1611446400 | 13.11
1/25/2021 | 1611532800 | 13.03
1/26/2021 | 1611619200 | 13.09
1/27/2021 | 1611705600 | 13.19
1/28/2021 | 1611792000 | 13.12
1/29/2021 | 1611878400 | 13.25
1/30/2021 | 1611964800 | 13.36
1/31/2021 | 1612051200 | 12.87
2/1/2021 | 1612137600 | 13.13
2/2/2021 | 1612224000 | 13.17
2/3/2021 | 1612310400 | 13.07

Reply:
Rebased to include the recent change that unified the two distinct return calculations that were present earlier. Block counts are now always sampled from archive data and not estimated, so the inaccuracy pointed out above is removed.

Reply:
Great work!